cura/Machines/QualityManager.py

```diff
@@ -258,40 +248,46 @@ def getQualityGroups(self, machine: "GlobalStack") -> dict:
                 # Also try to get the fallback material
                 material_type = extruder.material.getMetaDataEntry("material")
                 fallback_root_material_id = self._material_manager.getFallbackMaterialIdByMaterialType(material_type)
-                if fallback_root_material_id:
+                if fallback_root_material_id and root_material_id not in root_material_id_list:
                     root_material_id_list.append(fallback_root_material_id)
 
             # Here we construct a list of nodes we want to look for qualities with the highest priority first.
             # The use case is that, when we look for qualities for a machine, we first want to search in the following
             # order:
-            #   1. machine-variant-and-material-specific qualities if exist
-            #   2. machine-variant-specific qualities if exist
-            #   3. machine-material-specific qualities if exist
-            #   4. machine-specific qualities if exist
-            #   5. generic qualities if exist
+            #   1. machine-nozzle-buildplate-and-material-specific qualities if exist
+            #   2. machine-nozzle-and-material-specific qualities if exist
+            #   3. machine-nozzle-specific qualities if exist
+            #   4. machine-material-specific qualities if exist
+            #   5. machine-specific qualities if exist
+            #   6. generic qualities if exist
```

Comment:
Maybe an idea to put this in the top of the file since it's sort of the crux of adding build plates, right?